File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int BinaryCount(int x){
    string num = "";
    int count = 0;
    
    while(x>0){
        num = to_string(x%2) + num;
        x/=2;
    }
    
    for(int i=0;i<num.size();i++){
        if(num[i] == '1'){
            count++;
        }
    }
    
    return count;
}

int solution(int n) {
    int answer = n+1;
    int temp = BinaryCount(n);
    
    while(BinaryCount(answer) != temp){
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cs
using System;

class Solution 
{
    public int BinaryOneCount(int x){
        string binary = Convert.ToString(x,2);
        int count = 0;
        
        for(int i=0;i<binary.Length;i++){
            if(binary[i] == '1'){
                count++;
            }
        }     
        
        return count;
    }
    
    public int solution(int n) 
   {
        int answer = 0;
        
        int nCount = BinaryOneCount(n);
        
        while(true){
            n++;
            int tempCount = BinaryOneCount(n);
            
            if(nCount == tempCount){
                answer = n;
                break;
            }
        } 
        
        return answer;
    }
}